transmission.go

```diff
@@ -259,24 +252,32 @@ func (b *batchAgg) fireBatch(events []*Event) {
 	sd.Count("messages_sent", numEncoded)
 	defer resp.Body.Close()
 
-	// decode the responses
-	batchResponses := []Response{}
-	err = json.NewDecoder(resp.Body).Decode(&batchResponses)
-	if err != nil {
-		// if we can't decode the responses, just error out all of them
-		sd.Increment("response_decode_errors")
+	if resp.StatusCode != http.StatusOK {
+		sd.Increment("send_errors")
+		body, _ := ioutil.ReadAll(resp.Body)
```

Comment:
Hm, it strikes me as not impossible for this to return an `err`. I'm not really sure what to do about it (somehow add it as additional information in the error queue maybe?), but just swallowing the error feels a bit odd.